src/engine/contactreader.cpp

```diff
@@ -394,12 +399,12 @@ static void setValues(QContactOnlineAccount *detail, QSqlQuery *query, const int
 
     setValue(detail, T::FieldAccountUri     , query->value(offset + 0));
     // ignore lowerAccountUri
-    setValue(detail, T::FieldProtocol       , QVariant::fromValue<int>(OnlineAccount::protocol(query->value(offset + 2).toString())));
+    setValue(detail, T::FieldProtocol       , QVariant::fromValue<int>(query->value(offset + 2).toString().toInt()));
     setValue(detail, T::FieldServiceProvider, query->value(offset + 3));
     setValue(detail, T::FieldCapabilities   , stringListValue(query->value(offset + 4)));
 
     QStringList subTypeNames(query->value(offset + 5).toString().split(QLatin1Char(';'), QString::SkipEmptyParts));
```

Comment:
subTypeList seems to just call QString::number() on each string in the list, so subTypeNames could perhaps be renamed to subTypeValues?